pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -247,13 +249,29 @@ func (o *Options) Run() error {
 					break
 				}
 			}
+			if o.AllowNotRecommended {
+				for _, upgrade := range cv.Status.ConditionalUpdates {
+					if c := findCondition(upgrade.Conditions, "Recommended"); c != nil && c.Status != metav1.ConditionTrue {
+						if upgrade.Release.Version == o.To {
+							update = &configv1.Update{
+								Version: upgrade.Release.Version,
+								Image:   upgrade.Release.Image,
+							}
+							break
+						}
+					}
+				}
+			}
 			if update == nil {
 				if len(cv.Status.AvailableUpdates) == 0 {
 					if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 						return fmt.Errorf("Can't look up image for version %s. %v", o.To, c.Message)
 					}
 					return fmt.Errorf("No available updates, specify --to-image or wait for new updates to be available")
```

Comment:
We probably have to shard this out into a switch or something, because I see more cases now than we're currently handling:

* If `RetrievedUpdates!=True`, then we should complain about that.
* If there are no recommended or conditional updates, then `--to-image` or wait.
* If there are no recommended updates, but there are conditional updates, and...
  * `--allow-not-recommended`, so the user already knows they're living dangerously, and we just tell them that their requested `To` wasn't found, and maybe list the available conditional choices so they can easily fix typos.
  * no `--allow-not-recommended`, so `--to-image` or wait, but also possibly mention viewing with `--include-not-recommended` to see the conditional updates.
* If there are recommended updates but no conditional updates, then list the recommended choices so they can easily fix typos.
* If there are both recommended and conditional updates, and...
  * `--allow-not-recommended`, so the user already knows they're living dangerously, and we just tell them that their requested `To` wasn't found, and list the available choices (recommended and conditional) so they can easily fix typos.
  * no `--allow-not-recommended`, list the recommended choices so the can easily fix typos.

If that sounds sane, one way to simplify in this block would be to collect a `[]string` slice of choices as we walk above, so by the time we get down here we can use a `Join` to easily format the choices regardless of `--allow-not-recommended`-ness (or see that that choice set is empty).

Reply:
For `--to-image` there is a different code block below the current one. i.e. `if len(o.ToImage) > 0 {` which looks fine to me.

Reply:
So I trying to simplify 
```
	if update == nil {
				if len(cv.Status.AvailableUpdates) == 0 {
					if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
						return fmt.Errorf("Can't look up image for version %s. %v", o.To, c.Message)
					}
					return fmt.Errorf("No available updates, specify --to-image or wait for new updates to be available")
				}
				if o.AllowNotRecommended {
					return fmt.Errorf("The update %s is not one of the conditional or recommended updates", o.To)
				}
				return fmt.Errorf("The update %s is not one of the available updates: %s", o.To, strings.Join(versionStrings(cv.Status.AvailableUpdates), ", "))
			}
```

Reply:
PTAL https://github.com/openshift/oc/commit/7316d191bb03657921a4680fdb4a12106be33c81 and see if looks better.